aineko/extras/connectors/http_poller.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Extra module for connecting to an HTTP endpoint."""
+
+import json
+import time
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import requests
+from pydantic import BaseModel, field_validator
+
+from aineko import AbstractNode
+
+
+class ParamsHTTPPoller(BaseModel):
+    """Parameters for the HTTPPoller node.
+
+    Attributes:
+        timeout (int): The number of seconds to wait for the HTTP endpoint to
+            respond. Defaults to 10.
+        url (str): The HTTP URL to connect to.
+        headers (Optional[Dict[str, Any]]): A dictionary of headers to send to
+            the HTTP endpoint. Defaults to None.
+        data (Optional[Dict[str, Any]]): A dictionary of data to send to the
+            HTTP endpoint. Defaults to None.
+        params (Optional[Union[Dict[str, Any], List[tuple], bytes]]): A
+            dictionary, list of tuples, bytes, or file-like object to send in
+            the body of the HTTP request. Defaults to None.
+        json_ (Optional[Dict[str, Any]]): A JSON serializable Python object to
+            send in the body of the HTTP request. Defaults to None.
+        auth (Optional[Tuple[str, str]]): A tuple of username and password to
+            use for Basic HTTP authentication. Defaults to None.
+        poll_interval (float): The number of seconds to wait between polls.
+            Defaults to 5.0.
+        max_retries (int): The maximum number of times to retry connecting to
+            the HTTP endpoint. Defaults to -1.
+        metadata (Optional[Dict[str, Any]]): A dictionary of metadata to attach
+            to outgoing messages. Defaults to None.
+        retry_sleep (float): The number of seconds to wait between retries.
+            Defaults to 5.0.
+        success_codes (List[int]): A list of HTTP status codes that indicate
+            success. Defaults to
+                [200, 201, 202, 203, 204, 205, 206, 207, 208, 226].
```

Comment:
The type hints are not needed here. They will automatically be inferred on the docs.